client/src/main/java/model/Ship.java

```diff
@@ -37,21 +34,21 @@ boolean isShipDone() {
 
   void addMast(Mast mast) {
     if (!isShipDone()) {
-      masts[buildLength++] = mast;
-      positions.add(mast.positionAsInteger());
+      masts.add(mast);
+      buildLength++;
     }
   }
 
-  List<Position> mastsInShipPositions() {
-    return Arrays.stream(masts).map(Field::position).collect(Collectors.toList());
+  List<Position> positionsOfAllMastInShip() {
+    return masts.stream().map(Field::position).collect(Collectors.toList());
   }
 
-  public List<Integer> getPositions() {
-    return positions;
+  List<Integer> positionsOfAllMastInShipAsIntegers() {
```

Comment:
that's a weird method, but I am not judging

Reply:
why weird? :)

Reply:
Why do you need both positionsOfAllMastInShip() and positionsOfAllMastInShipAsIntegers() ?